serializers/shared/src/main/java/io/pravega/schemaregistry/serializer/shared/impl/AbstractDeserializer.java

```diff
@@ -93,8 +93,17 @@ public T deserialize(ByteBuffer data) {
                 decoded = decoders.decode(encodingInfo.getCodecType(), data);
             }
 
-            inputStream = new ByteArrayInputStream(decoded.array(), 
-                    decoded.arrayOffset() + decoded.position(), decoded.remaining());
+            byte[] b;
+            if (decoded.hasArray()) {
```

Comment:
Curious, why is this check needed? Do you expect this could also be a native byte buffer without a backing array?

Reply:
this class shouldnt make any assumptions as the contract is that it would receive a bytebuffer. which can be a direct buffer too